mat64/io.go

```diff
@@ -81,18 +87,24 @@ func (m *Dense) UnmarshalBinary(data []byte) error {
 		panic("mat64: unmarshal into non-zero matrix")
 	}
 
+	var tmp [8]byte
 	buf := bytes.NewReader(data)
-	var rows int64
-	err := binary.Read(buf, binary.LittleEndian, &rows)
+	_, err := buf.Read(tmp[:])
```

Comment:
ie: comparing `io.ReadFull` with the first version of `readFull` that also checks the number of bytes read:

```
name                    old time/op    new time/op    delta
UnmarshalDense10-4         778ns ± 3%     683ns ± 1%  -12.16%          (p=0.008 n=5+5)
UnmarshalDense100-4       5.13µs ± 2%    4.11µs ± 1%  -19.94%          (p=0.008 n=5+5)
UnmarshalDense1000-4      48.4µs ± 1%    37.6µs ± 1%  -22.19%          (p=0.008 n=5+5)
UnmarshalDense10000-4      464µs ± 4%     366µs ± 1%  -21.17%          (p=0.008 n=5+5)
UnmarshalVector10-4        733ns ± 1%     638ns ± 1%  -13.01%          (p=0.008 n=5+5)
UnmarshalVector100-4      5.17µs ± 2%    4.10µs ± 2%  -20.65%          (p=0.008 n=5+5)
UnmarshalVector1000-4     49.0µs ± 1%    37.9µs ± 1%  -22.65%          (p=0.008 n=5+5)
UnmarshalVector10000-4     477µs ± 7%     368µs ± 2%  -22.84%          (p=0.008 n=5+5)
```

(the other numbers are unchanged)

so, I'd err on going with the `readFull` version with the check on `n`.